module/executiondatasync/provider/provider.go

```diff
@@ -115,6 +115,9 @@ func (p *Provider) storeBlobs(parent context.Context, blockHeight uint64, blobCh
 
 // Provide adds the block execution data for a newly executed (generally not sealed or finalized) block to the blob store for distribution using Bitswap.
 // It computes and returns the root CID of the execution data blob tree.
+// This function returns once the root CID has been computed, and all blobs are queued for storage
+// in the Bitswap Blobstore. Storage operations are performed asynchronously and may continue after
+// this function returns.
```

Comment:
```suggestion
// This function returns once the root CID has been computed, and all blobs are successfully stored
// in the Bitswap Blobstore.
```
Adjusting my original suggestion here, because it was based on an incorrect understanding of the behaviour.